src/rpc/rawtransaction.cpp

```diff
@@ -213,6 +214,142 @@ static UniValue getrawtransaction(const JSONRPCRequest& request)
+            "   }\n"
+            " ],\n"
+            " ]\n"
+                },
+            },
+            RPCExamples{
+                    HelpExampleCli("searchrawtransactions", "\"address\"")
+        + HelpExampleCli("searchrawtransactions", "\"address\" true")
+            },
+        }.ToString());
+
+    CScript scriptPubKey;
+    const CTxDestination dest = DecodeDestination(request.params[0].get_str());
+    if(IsValidDestination(dest)) {
+        scriptPubKey = GetScriptForDestination(dest);
+    } else if(IsHex(request.params[0].get_str())) {
+        std::vector<unsigned char> data(ParseHex(request.params[0].get_str()));
+        scriptPubKey = CScript(data.begin(), data.end());
+    } else {
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid Bitcoin address or script");
+    }
+
+    // Accept either a bool (true) or a num (>=1) to indicate verbose output.
+    bool verbose = false;
+    if (!request.params[1].isNull()) {
+        verbose = request.params[1].isNum() ? (request.params[1].get_int() != 0) : request.params[1].get_bool();
+    }
+
+    int count = 100;
+    if (request.params.size() > 2) {
+        if (!request.params[2].isNum()) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, argument 3 must be an integer");
+        }
+        count = request.params[2].get_int();
+    }
+
+    if (!g_addr_index) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "This RPC requires -addrindex to be enabled.");
+    }
+
+    bool addrindex_ready = g_addr_index->BlockUntilSyncedToCurrentChain();
+
+    UniValue ret(UniValue::VARR);
+
+    std::vector<std::pair<COutPoint, std::pair<CTransactionRef, uint256>>> spends_result;
+    std::vector<std::pair<COutPoint, std::pair<CTransactionRef, uint256>>> creations_result;
+    if (!g_addr_index->FindTxsByScript(scriptPubKey, spends_result, creations_result)) {
+        if (!addrindex_ready) {
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY,"Transactions with given address not found. Blockchain transactions are still in the process of being indexed");
+        }
+        return ret;
+    }
+
+    UniValue spends(UniValue::VARR);
+    UniValue creations(UniValue::VARR);
+
+    auto spends_it = spends_result.begin();
+    while (spends_it != spends_result.end() && count--) {
+        const auto& spend = *spends_it;
+        UniValue tx_val(UniValue::VOBJ);
+
+        UniValue outpoint_val(UniValue::VOBJ);
+        outpoint_val.pushKV("txid", spend.first.hash.GetHex());
+        outpoint_val.pushKV("n", static_cast<int>(spend.first.n));
+
+        tx_val.pushKV("spent_outpoint", outpoint_val);
+        if (verbose) {
+            TxToJSON(*(spend.second.first), spend.second.second, tx_val);
+        } else {
+            std::string hex_tx = EncodeHexTx(*(spend.second.first), RPCSerializationFlags());
+            tx_val.pushKV("hex", hex_tx);
+        }
+        spends.push_back(tx_val);
+        ++spends_it;
+    }
```

Comment:
If there are more spends than `count` then no creates will be returned. Ideally the spends and creates will be returned in order, rather than all spends first.